src/core/server/index.ts

```diff
@@ -221,7 +224,19 @@ class Server {
     await this.i18n.load();
 
     // Setup MongoDB.
-    this.mongo = await createMongoDB(config);
+    const mongoURI = config.get("mongodb");
+    let archiveMongoURI = config.get("mongodbArchive");
+    // if the archive URI isn't specified, but the regular mongo uri
+    // is, then use the same db for both mongo instances
+    if (
+      archiveMongoURI === config.default("mongodbArchive") &&
+      mongoURI !== config.default("mongodb")
+    ) {
+      archiveMongoURI = mongoURI;
+    }
+
+    this.mongo = await createMongoDB(mongoURI);
+    this.archive = await createMongoDB(archiveMongoURI);
```

Comment:
I would rather see this be a bit different! Right now, if the same URL is for both the archive and the mongo instance, we actually create 2X the connections to the same instance! Instead, I'd suggest the following logic:

```suggestion
    this.mongo = await createMongoDB(mongoURI);

    const mongoURI = config.get("mongodb");
    let archiveMongoURI = config.get("mongodbArchive");
    // if the archive URI isn't specified, but the regular mongo uri
    // is, then use the same db for both mongo instances
    if (
      archiveMongoURI === config.default("mongodbArchive") &&
      mongoURI !== config.default("mongodb")
    ) {
      this.archive = this.mongo;
    } else {
      this.archive = await createMongoDB(archiveMongoURI);
    }
```